satpy/readers/__init__.py

```diff
@@ -538,3 +539,61 @@ def _get_reader_kwargs(reader, reader_kwargs):
         reader_kwargs_without_filter[k].pop('filter_parameters', None)
 
     return (reader_kwargs, reader_kwargs_without_filter)
+
+
+@total_ordering
+class FSFile(os.PathLike):
+    """Implementation of a PathLike file object, that can be opened.
```

Comment:
I'm confused by this line.  Is it a file object or a path object?  If it's a file object it's already opened?  I think `os.PathLike` represents a path, not an open file.

Reply:
It is a path indeed, but can be opened to return a file object.